src/main/java/com/fundynamic/d2tm/game/event/ViewportMovementListener.java

```diff
@@ -32,17 +35,24 @@ public void mouseClicked(int button, int x, int y, int clickCount) {
         if (clickCount == 1) {
             if (button == Input.MOUSE_LEFT_BUTTON) {
                 if (mouse.hoversOverSelectableEntity()) {
-                    mouse.deselectStructure(); // deselect any previously selected structure
-                    mouse.selectStructure();
+                    mouse.deselectEntity();
+                    mouse.selectEntity();
+
+                    // add some fun so we place units/structures depending on what we selected
+                    if (mouse.getLastSelectedEntity() instanceof Structure) {
+                        entityType = EntityRepository.EntityType.STRUCTURE;
+                    } else {
+                        entityType = EntityRepository.EntityType.UNIT;
```

Comment:
this is some quick code to place stuff on the screen. Eventually we don't place `Units`, probably only `Structures`.